packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -173,11 +173,12 @@ function basicStateReducer<S>(state: S, action: BasicStateAction<S>): S {
 }
 
 class ReactShallowRenderer {
-  static createRenderer = function() {
-    return new ReactShallowRenderer();
+  static createRenderer = function(useLifecycleMethods) {
+    return new ReactShallowRenderer(useLifecycleMethods);
   };
 
-  constructor() {
+  constructor(useLifecycleMethods = false) {
```

Comment:
The way this option is passed can be changed. I just did this for now because it was easy.